django/db/models/sql/compiler.py

```diff
@@ -235,7 +204,9 @@ def collapse_group_by(self, expressions, having):
             expressions = [
                 expr
                 for expr in expressions
-                if expr in pks or getattr(expr, "alias", None) not in aliases
+                if expr in pks
+                or expr in having
```

Comment:
Even if not required on Postgres I figured the simplicity tradeoff this brings to `collapse_group_by` was worth it.